docs/Makefile

```diff
@@ -60,7 +61,10 @@ rest-api: source/_static/rest-api/index.html
 source/_static/rest-api/index.html: rest-api.yml node_modules
 	npm run rest-api
 
-html: rest-api
+generate:
+	python3 generator.py
+
+html: rest-api generate
```

Comment:
Putting this in the makefile means it likely won't get built on readthedocs (which calls sphinx directly, not via the makefile, unless something has changed)

Maybe this should be written as a sphinx extension instead?

Reply:
I'm afraid I don't have any experience with sphinx/sphinx-extensions.

I did find a [way to invoke](https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx-core-events) our python script via the `setup()` method in `conf.py`. 

Not sure which approach is better. In case you prefer to have an extension, it'd be great if you could point me towards some resource/reference (for a noob).